tests/common/apiversion.go

```diff
@@ -0,0 +1,58 @@
+package common
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+)
+
+var (
+	ingressPath = "/01_keycloak_helmchart/keycloak/templates/ingress.yaml"
+)
+
+func IngressApiVersionTest(t *testing.T, basepath string) {
+	// Setup decoding
+	subject := &networkingv1.Ingress{}
+	scheme := NewSchemeWithDefault(t)
+	require.NoError(t, networkingv1.AddToScheme(scheme))
+
+	// Read file
+	data, err := ioutil.ReadFile(basepath + ingressPath)
+	require.NoError(t, err)
+
+	// Decode object
+	kind := subject.GetObjectKind().GroupVersionKind()
+	decode, _, err := serializer.NewCodecFactory(scheme).UniversalDeserializer().Decode(data, &kind, subject)
+	require.NoError(t, err)
+	ingress, ok := decode.(*networkingv1.Ingress)
+	assert.True(t, ok, "Expected networking.k8s.io/v1 Ingress object")
+	if ok {
+		assert.Equal(t, "keycloak", ingress.Name)
+	}
```

Comment:
This can be simplified as well
```suggestion
func IngressApiVersionTest(t *testing.T, basepath string) {
	subject := &networkingv1.Ingress{}
	scheme := NewSchemeWithDefault(t)
	ingress := DecodeWithSchema(t, basepath+ingressPath, subject, scheme).(*networkingv1.Ingress)
	assert.Equal(t, "keycloak", ingress.Name)
```